### PR TITLE
PHOENIX-6378 Unbundle sqlline from phoenix-client

### PR DESCRIPTION
**Pull request by richardantal**
Change-Id: I443593c18d412c4ccd2e137c6b2809863bc71bc8

### CODE REVIEW

File: phoenix-assembly/pom.xml [hunk 1]
```diff
@@ -132,6 +132,10 @@
       <groupId>org.apache.phoenix</groupId>
       <artifactId>phoenix-client-${hbase.suffix}</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.apache.phoenix</groupId>
+      <artifactId>phoenix-client-embedded-${hbase.suffix}</artifactId>
+    </dependency>
```
Comment (stoty) [inline phoenix-assembly/pom.xml:138]: I don't think we should to add the standard (old) client to the assembly after this change.

File: phoenix-assembly/src/build/package-to-tar-all.xml [hunk 2]
```diff
@@ -30,8 +30,23 @@
   </formats>
   <includeBaseDirectory>true</includeBaseDirectory>
 
+  <!-- Components that we don't want in jars that are used with other libraries, but we want for a standalone client -->
+  <dependencySets>
+    <dependencySet>
+      <!-- Unpack all the dependencies to class files, since java doesn't support
+        jar of jars for running -->
+      <unpack>false</unpack>
+      <!-- save these dependencies to the top-level -->
```
Comment (stoty) [inline phoenix-assembly/src/build/package-to-tar-all.xml:39]: This comment is just false

File: phoenix-assembly/src/build/package-to-tar-all.xml [hunk 3]
```diff
@@ -30,8 +30,23 @@
   </formats>
   <includeBaseDirectory>true</includeBaseDirectory>
 
+  <!-- Components that we don't want in jars that are used with other libraries, but we want for a standalone client -->
+  <dependencySets>
+    <dependencySet>
+      <!-- Unpack all the dependencies to class files, since java doesn't support
```
Comment (stoty) [inline phoenix-assembly/src/build/package-to-tar-all.xml:36]: I don't think this comment is relevant to anything anymore.

File: phoenix-client-parent/phoenix-client/pom.xml [hunk 4]
```diff
@@ -85,10 +85,5 @@
       <artifactId>log4j</artifactId>
       <scope>runtime</scope>
     </dependency>
-    <dependency>
```
Comment (stoty) [inline phoenix-client-parent/phoenix-client/pom.xml:88]: I think that we'd best keep the old client unchanged for backwards compatibility reasons.

File: phoenix-core/pom.xml [hunk 5]
```diff
@@ -523,6 +523,13 @@
       <artifactId>hamcrest-core</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
```
Comment (stoty) [inline phoenix-core/pom.xml:526]: Why is this needed ?
Ideally Phoenix-core doesn't have anything to do with sqlline.

File: phoenix-client-parent/phoenix-client-embedded/pom.xml [hunk 6]
```diff
@@ -81,10 +81,5 @@
       <artifactId>phoenix-hbase-compat-${hbase.compat.version}</artifactId>
       <optional>false</optional>
     </dependency>
-    <dependency>
```
Comment (stoty) [inline phoenix-client-parent/phoenix-client-embedded/pom.xml:84]: I thought that we are already not adding sqlline here.

Just noting that we change sqlline-embedded too.

File: bin/phoenix_utils.py [hunk 7]
```diff
@@ -77,10 +77,13 @@ def findClasspath(command_name):
     return tryDecode(subprocess.Popen(command, shell=True, stdout=subprocess.PIPE).stdout.read())
 
 def setPath():
-    PHOENIX_CLIENT_JAR_PATTERN = "phoenix-client-hbase-*[!s].jar"
+    PHOENIX_CLIENT_EMBEDDED_JAR_PATTERN = "phoenix-client-embedded-hbase-*[!s].jar"
```
Comment (stoty) [inline bin/phoenix_utils.py:80]: phoenix-client-embedded is missing the SLF4J backend as well.
It should handled similarly to the sqlline JAR, with loading it from /lib by default, but having an option to override (and use a different backed JAR)

**Review by stoty — CHANGES_REQUESTED** [review]

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   5m 41s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   5m 12s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  11m 24s |  master passed  |
| +0 |  hbaserecompile  |  23m 12s |  HBase recompiled.  |
| +1 :green_heart: |  compile  |   1m 30s |  master passed  |
| +1 :green_heart: |  javadoc  |   2m 59s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   1m 14s |  root in the patch failed.  |
| +0 |  hbaserecompile  |   9m 13s |  HBase recompiled.  |
| -1 :x: |  compile  |   1m 30s |  root in the patch failed.  |
| -1 :x: |  javac  |   1m 30s |  root in the patch failed.  |
| -1 :x: |  pylint  |   0m  8s |  The patch generated 16 new + 115 unchanged - 10 fixed = 131 total (was 125)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m 11s |  The patch has no ill-formed XML file.  |
| -1 :x: |  javadoc  |   1m 20s |  root in the patch failed.  |
| -1 :x: |  javadoc  |   0m 14s |  phoenix-assembly in the patch failed.  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 113m 15s |  root in the patch failed.  |
| -1 :x: |  asflicense  |   3m 49s |  The patch generated 614 ASF License warnings.  |
|  |   | 168m  6s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.end2end.PermissionNSEnabledWithCustomAccessControllerIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1239 |
| Optional Tests | dupname asflicense pylint javac javadoc unit xml compile |
| uname | Linux 25a3da6b7e19 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 3231aad |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| mvninstall | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/1/artifact/yetus-general-check/output/patch-mvninstall-root.txt |
| compile | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/1/artifact/yetus-general-check/output/patch-compile-root.txt |
| javac | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/1/artifact/yetus-general-check/output/patch-compile-root.txt |
| pylint | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/1/artifact/yetus-general-check/output/diff-patch-pylint.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/1/artifact/yetus-general-check/output/patch-javadoc-root.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/1/artifact/yetus-general-check/output/patch-javadoc-phoenix-assembly.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/1/artifact/yetus-general-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/1/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/1/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 15434 (vs. ulimit of 30000) |
| modules | C: phoenix-core phoenix-client-parent/phoenix-client phoenix-client-parent/phoenix-client-embedded . phoenix-assembly U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/1/console |
| versions | git=2.7.4 maven=3.3.9 pylint=2.4.4 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   5m 38s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  12m 43s |  master passed  |
| +0 |  hbaserecompile  |  25m 47s |  HBase recompiled.  |
| +1 :green_heart: |  compile  |   1m 35s |  master passed  |
| +1 :green_heart: |  javadoc  |   3m 16s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 23s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |  10m  1s |  root in the patch failed.  |
| +0 |  hbaserecompile  |  17m 49s |  HBase recompiled.  |
| -1 :x: |  compile  |   1m 35s |  root in the patch failed.  |
| -1 :x: |  javac  |   1m 35s |  root in the patch failed.  |
| -1 :x: |  pylint  |   0m  7s |  The patch generated 22 new + 115 unchanged - 10 fixed = 137 total (was 125)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m 11s |  The patch has no ill-formed XML file.  |
| -1 :x: |  javadoc  |   1m 26s |  root in the patch failed.  |
| -1 :x: |  javadoc  |   0m 13s |  phoenix-assembly in the patch failed.  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 115m 28s |  root in the patch failed.  |
| -1 :x: |  asflicense  |   3m 40s |  The patch generated 614 ASF License warnings.  |
|  |   | 176m 54s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1239 |
| Optional Tests | dupname asflicense pylint javac javadoc unit xml compile |
| uname | Linux 6a0da1a9988b 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 3231aad |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| mvninstall | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/2/artifact/yetus-general-check/output/patch-mvninstall-root.txt |
| compile | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/2/artifact/yetus-general-check/output/patch-compile-root.txt |
| javac | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/2/artifact/yetus-general-check/output/patch-compile-root.txt |
| pylint | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/2/artifact/yetus-general-check/output/diff-patch-pylint.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/2/artifact/yetus-general-check/output/patch-javadoc-root.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/2/artifact/yetus-general-check/output/patch-javadoc-phoenix-assembly.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/2/artifact/yetus-general-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/2/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/2/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 15461 (vs. ulimit of 30000) |
| modules | C: phoenix-core phoenix-client-parent/phoenix-client phoenix-client-parent/phoenix-client-embedded . phoenix-assembly U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/2/console |
| versions | git=2.7.4 maven=3.3.9 pylint=2.4.4 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 55s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   5m 14s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  11m  7s |  master passed  |
| +0 |  hbaserecompile  |  23m  6s |  HBase recompiled.  |
| +1 :green_heart: |  compile  |   1m 32s |  master passed  |
| +1 :green_heart: |  javadoc  |   3m  3s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   7m 54s |  root in the patch failed.  |
| +0 |  hbaserecompile  |  14m 35s |  HBase recompiled.  |
| -1 :x: |  compile  |   1m 29s |  root in the patch failed.  |
| -1 :x: |  javac  |   1m 29s |  root in the patch failed.  |
| -1 :x: |  pylint  |   0m  8s |  The patch generated 22 new + 115 unchanged - 10 fixed = 137 total (was 125)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m 11s |  The patch has no ill-formed XML file.  |
| -1 :x: |  javadoc  |   1m 21s |  root in the patch failed.  |
| -1 :x: |  javadoc  |   0m 14s |  phoenix-assembly in the patch failed.  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 116m 13s |  root in the patch failed.  |
| -1 :x: |  asflicense  |   3m  2s |  The patch generated 614 ASF License warnings.  |
|  |   | 170m 48s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1239 |
| Optional Tests | dupname asflicense pylint javac javadoc unit xml compile |
| uname | Linux e2cfb2bde495 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 63cbb11 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| mvninstall | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/3/artifact/yetus-general-check/output/patch-mvninstall-root.txt |
| compile | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/3/artifact/yetus-general-check/output/patch-compile-root.txt |
| javac | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/3/artifact/yetus-general-check/output/patch-compile-root.txt |
| pylint | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/3/artifact/yetus-general-check/output/diff-patch-pylint.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/3/artifact/yetus-general-check/output/patch-javadoc-root.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/3/artifact/yetus-general-check/output/patch-javadoc-phoenix-assembly.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/3/artifact/yetus-general-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/3/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/3/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 15400 (vs. ulimit of 30000) |
| modules | C: phoenix-core phoenix-client-parent/phoenix-client phoenix-client-parent/phoenix-client-embedded . phoenix-assembly U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/3/console |
| versions | git=2.7.4 maven=3.3.9 pylint=2.4.4 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
Could you update the ticket title and commit message ?
Something like 
Unbunldle sqqline from phoenix-client-embedded, and use it in sqlline.py

File: bin/phoenix_utils.py [hunk 8]
```diff
@@ -77,10 +77,16 @@ def findClasspath(command_name):
...
     PHOENIX_TRACESERVER_JAR_PATTERN = "phoenix-tracing-webapp-*-runnable.jar"
     PHOENIX_TESTS_JAR_PATTERN = "phoenix-core-*-tests*.jar"
     PHOENIX_PHERF_JAR_PATTERN = "phoenix-pherf-*[!s].jar"
+    SLF4J_JAR_PATTERN = "slf4j-log4j12-*[!s].jar"
```
Comment (stoty) [inline bin/phoenix_utils.py:84]: Please change the slf4j* variable names to slf4j_log4j12_* for clarity.

File: bin/sqlline.py [hunk 9]
```diff
@@ -108,7 +108,9 @@ def kill_child():
     colorSetting = "false"
 
 java_cmd = java + ' $PHOENIX_OPTS ' + \
-    ' -cp "' + hbase_config_path + os.pathsep + phoenix_utils.hbase_conf_dir + os.pathsep + phoenix_utils.phoenix_client_jar + \
+    ' -cp "' + phoenix_utils.sqlline_with_deps_jar + os.pathsep + hbase_config_path + os.pathsep + \
+    phoenix_utils.slf4j_jar + os.pathsep + \
```
Comment (stoty) [inline bin/sqlline.py:112]: this variable name hasn't been updated.

File: bin/phoenix_utils.py [hunk 10]
```diff
@@ -176,6 +182,16 @@ def setPath():
     if phoenix_pherf_jar == "":
         phoenix_pherf_jar = findFileInPathWithoutRecursion(PHOENIX_PHERF_JAR_PATTERN, os.path.join(current_dir, ".."))
 
+    global sqlline_with_deps_jar
+    sqlline_with_deps_jar = os.environ.get(OVERRIDE_SQLLINE)
+    if sqlline_with_deps_jar is None or sqlline_with_deps_jar == "":
+        sqlline_with_deps_jar = findFileInPathWithoutRecursion(SQLLINE_WITH_DEPS_PATTERN, os.path.join(current_dir, "..","lib"))
+
+    global slf4j_jar
```
Comment (stoty) [inline bin/phoenix_utils.py:190]: this variable name hasn't been updated

File: bin/phoenix_utils.py [hunk 11]
```diff
@@ -216,3 +232,4 @@ def common_sqlline_args(parser):
     print("phoenix_loadbalancer_jar:", phoenix_loadbalancer_jar)
     print("phoenix_thin_client_jar:", phoenix_thin_client_jar)
     print("hadoop_classpath:", hadoop_classpath)
+    print("sqlline_with_deps_jar", sqlline_with_deps_jar)
```
Comment (stoty) [inline bin/phoenix_utils.py:235]: We should log the slf4j_log4j12 jar name too.

**Review by stoty — CHANGES_REQUESTED** [review]

File: bin/phoenix_utils.py [hunk 12]
```diff
@@ -77,10 +77,16 @@ def findClasspath(command_name):
...
-    PHOENIX_CLIENT_JAR_PATTERN = "phoenix-client-hbase-*[!s].jar"
+    PHOENIX_CLIENT_EMBEDDED_JAR_PATTERN = "phoenix-client-embedded-hbase-*[!s].jar"
     PHOENIX_TRACESERVER_JAR_PATTERN = "phoenix-tracing-webapp-*-runnable.jar"
     PHOENIX_TESTS_JAR_PATTERN = "phoenix-core-*-tests*.jar"
     PHOENIX_PHERF_JAR_PATTERN = "phoenix-pherf-*[!s].jar"
+    SLF4J_LOG4J12_JAR_PATTERN = "slf4j-log4j12-*[!s].jar"
+    SQLLINE_WITH_DEPS_PATTERN = "sqlline-*-jar-with-dependencies.jar"
+
+
+    OVERRIDE_SLF4J = "OVERRIDE_SLF4J_JAR_LOCATION"
```
Comment (stoty) [inline bin/phoenix_utils.py:88]: On second thought, we should call it SLF4J_BACKEND_JAR instead of SLF4J_LOG4J12_JAR.
and apply it everywhere, including the name of the OVERRIDE system property.

**Review by stoty — APPROVED** [review]
+1 LGTM

**Comment by richardantal** [comment]
Thanks @stoty for the review.

I squashed the commits to 1, before merging.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   5m 57s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  11m 23s |  master passed  |
| +0 |  hbaserecompile  |  24m  5s |  HBase recompiled.  |
| +1 :green_heart: |  compile  |   1m 32s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m 58s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   8m  1s |  root in the patch failed.  |
| +0 |  hbaserecompile  |  14m 49s |  HBase recompiled.  |
| -1 :x: |  compile  |   1m 30s |  root in the patch failed.  |
| -1 :x: |  javac  |   1m 30s |  root in the patch failed.  |
| -1 :x: |  pylint  |   0m  8s |  The patch generated 22 new + 115 unchanged - 10 fixed = 137 total (was 125)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  8s |  The patch has no ill-formed XML file.  |
| -1 :x: |  javadoc  |   1m 23s |  root in the patch failed.  |
| -1 :x: |  javadoc  |   0m 14s |  phoenix-assembly in the patch failed.  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 119m 14s |  root in the patch failed.  |
| -1 :x: |  asflicense  |   2m 21s |  The patch generated 614 ASF License warnings.  |
|  |   | 171m 32s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1239 |
| Optional Tests | dupname asflicense pylint javac javadoc unit xml compile |
| uname | Linux 911b11d27d92 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 5c94168 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| mvninstall | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/4/artifact/yetus-general-check/output/patch-mvninstall-root.txt |
| compile | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/4/artifact/yetus-general-check/output/patch-compile-root.txt |
| javac | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/4/artifact/yetus-general-check/output/patch-compile-root.txt |
| pylint | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/4/artifact/yetus-general-check/output/diff-patch-pylint.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/4/artifact/yetus-general-check/output/patch-javadoc-root.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/4/artifact/yetus-general-check/output/patch-javadoc-phoenix-assembly.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/4/artifact/yetus-general-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/4/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/4/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 12800 (vs. ulimit of 30000) |
| modules | C: phoenix-client-parent/phoenix-client-embedded . phoenix-assembly U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1239/4/console |
| versions | git=2.7.4 maven=3.3.9 pylint=2.4.4 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.
